test/RightsService.Business.UnitTests/Commands/Role/GetRoleCommandTests.cs

```diff
@@ -1,43 +1,269 @@
+                _mocker.GetMock<IRoleInfoMapper>().Object,
+                _mocker.GetMock<IUserInfoMapper>().Object,
+                _mocker.GetMock<IRightResponseMapper>().Object,
+                _mocker.GetMock<IRequestClient<IGetUsersDataRequest>>().Object
+            );
         }
 
         [SetUp]
         public void SetUp()
         {
-            _mocker = new AutoMocker();
-            _command = _mocker.CreateInstance<IGetRoleCommand>();
+            _operationResultGetUsersData.Reset();
+            _mocker.GetMock<ILogger<GetRoleCommand>>();
+            _mocker.GetMock<IRoleRepository>();
+            _mocker.GetMock<IRightRepository>();
+            _mocker.GetMock<IRoleInfoMapper>();
+            _mocker.GetMock<IUserInfoMapper>();
+            _mocker.GetMock<IRightResponseMapper>();
+            _mocker.GetMock<IRequestClient<IGetUsersDataRequest>>();
+        }
+
+        [Test]
+        public void ShouldThrowNotFoundExceptionWhedRoleIdIsInvalid()
+        {
+            _mocker
+                .Setup<IRoleRepository, DbRole>(r => r.Get(It.IsAny<Guid>()))
+                .Throws(new NotFoundException());
+
+            Assert.Throws<NotFoundException>(() => _command.Execute(Guid.NewGuid()));
+        }
+
+        [Test]
+        public void ShouldThrowArgumentNullExceptionWhenRightIsNull()
+        {
+            DbRole roleWithNullRight = new DbRole()
+            {
+                Rights = new List<DbRoleRight>()
+                {
+                    new DbRoleRight() { Right = null }
+                }
+            };
+
+            _mocker
+                .Setup<IRoleRepository, DbRole>(r => r.Get(It.IsAny<Guid>()))
+                .Returns(roleWithNullRight);
+
+            _mocker
+                .Setup<IRightResponseMapper, RightResponse>(rrm => rrm.Map(It.Is<DbRight>(value => value == null)))
+                .Throws(new ArgumentNullException());
+
+            Assert.Throws<ArgumentNullException>(() => _command.Execute(Guid.NewGuid()));
         }
 
-        // TODO add tests
+        [Test]
+        public void ShouldGetAnErrorWhenConsumerReturnWrongUserData()
+        {
+            Guid invalidUserId = Guid.NewGuid();
+            int expectedErrorsCount = 1;
+
+            _mocker
+                .Setup<IRoleRepository, DbRole>(r => r.Get(It.IsAny<Guid>()))
+                .Returns(_dbRole);
+
+            _mocker
+                .Setup<IRightResponseMapper, RightResponse>(m => m.Map(It.IsNotNull<DbRight>()))
+                .Returns(_rightResponse);
+
+            _operationResultGetUsersData.Setup(r => r.Message.IsSuccess).Returns(true);
+            _operationResultGetUsersData.Setup(r => r.Message.Errors).Returns(new List<string>());
+            _operationResultGetUsersData
+                .Setup(r => r.Message.Body.UsersData)
+                .Returns(new List<UserData>() 
+                {
+                    // consumer return wrong UserData with invalid Id
```

Comment:
how i remember, we believe that other services is correct. it is necessary to clarify whether such checks are necessary

Reply:
Summon @MariyaSin

Reply:
removed test